quickstart/quickstart.Dockerfile

```diff
@@ -0,0 +1,87 @@
+# ---------------------------------------------
+# Docker Stage: Base builder used for both parsec service and tools
+FROM rust:latest AS base_builder
+
+RUN apt update -y && \
+    apt install -y llvm-dev libclang-dev clang cmake jq
+
+## Track various build environment things we may want to use throughout
+WORKDIR /build-env
+RUN echo "$(uname | awk '{print tolower($0)}')" > /build-env/os
+RUN echo "$(arch)" > /build-env/arch
+
+# ---------------------------------------------
+# Docker Stage: Temporary stage to help dependency caching
+FROM base_builder AS parsec_service_scratch
+
+## Copy everything in
+COPY . /parsec-service
+WORKDIR /parsec-service
+
+# This just adds cargo dependencies to the scratch stage so that we don't need to
+# download them each time the builder runs.
+RUN cargo fetch
+
+# Save the current parsec version as defined by cargo
+RUN echo "$(cargo metadata --format-version=1 --no-deps --offline | jq -r '.packages[0].version')" > /build-env/parsec-version
+
+# ---------------------------------------------
+# Docker Stage: Executes the build of the Parsec Service
+FROM parsec_service_scratch AS parsec_service_builder
+
+## Run the actual build
+RUN cargo build --release --features mbed-crypto-provider
+
+# ---------------------------------------------
+# Docker Stage: Executes the build of the Parsec Tool
+FROM base_builder AS parsec_tool_builder
+
+RUN git clone https://github.com/parallaxsecond/parsec-tool /parsec-tool
+WORKDIR /parsec-tool
+RUN git checkout $(git tag --sort=committerdate | tail -1)
+RUN cargo build --release
+
+# ---------------------------------------------
+# Docker Stage: Extracts build results from previous stages and adds in quickstart configs
+FROM base_builder AS layout
+
+## Add the built binaries into the image and make available on PATH
+COPY --from=parsec_service_builder /parsec-service/target/release/parsec /parsec/bin/parsec
+COPY --from=parsec_tool_builder /parsec-tool/target/release/parsec-tool /parsec/bin/parsec-tool
+
+## Create and configure a starting directory for quickstart operations
+WORKDIR /parsec/quickstart
+COPY quickstart/config.toml /parsec/quickstart/config.toml
+COPY --from=parsec_tool_builder /parsec-tool/tests/parsec-cli-tests.sh /parsec/quickstart/parsec-cli-tests.sh
```

Comment:
I think it's worth constructing file including useful build information for both the parsec server and tool. It could include things like
* Parsec and tool version, git commit hash, and git tag (if available)
* Parsec and tool build dependencies as generated by `cargo tree`
* Build OS and Arch

Reply:
The latest commit now generates and includes in `/parsec/quickstart` a file with build details.

Example
```
----------------------------------------
-- Parsec Quickstart Build Details
----------------------------------------
OS: linux
Architecture: x86_64
Rust: rustc 1.66.0 (69f9c33d7 2022-12-12)
Cargo: cargo 1.66.0 (d65d197ad 2022-11-15)

----------------------------------------
-- Parsec Service
----------------------------------------
Version: 1.1.0
Commit Hash: 4b89d8999b172925fe0cb2aafdcce068273ce01d
Dependencies:
parsec-service v1.1.0 (/parsec-service)
├── anyhow v1.0.64
├── base64 v0.13.0
├── bincode v1.3.3
│   └── serde v1.0.144
│       └── serde_derive v1.0.144 (proc-macro)
........

----------------------------------------
-- Parsec Tool
----------------------------------------
Version: 0.5.4
Commit Hash: 1dc8109ef77f556167fdfc9a57847b74170956ea
Dependencies:
parsec-tool v0.5.4 (/parsec-tool)
├── ansi_term v0.12.1
├── atty v0.2.14
│   └── libc v0.2.132
├── base64 v0.13.0
├── clap v2.34.0
│   ├── ansi_term v0.12.1
│   ├── atty v0.2.14 (*)
........
```

Reply:
Nice, thanks for adding this :)